src/main/scala/org/apache/spark/sql/TiStrategy.scala

```diff
@@ -180,11 +180,11 @@ class TiStrategy(context: SQLContext) extends Strategy with Logging {
   // and then we don't return (don't planLater) and plan the remaining all at once
   private def doPlan(source: TiDBRelation, plan: LogicalPlan): Seq[SparkPlan] = {
 
-    val aliasMap = mutable.HashMap[Expression, Alias]()
+    val aliasMap = mutable.HashMap[(String,Attribute), Alias]()
     val avgPushdownRewriteMap = mutable.HashMap[ExprId, List[AggregateExpression]]()
     val avgFinalRewriteMap = mutable.HashMap[ExprId, List[AggregateExpression]]()
 
-    def toAlias(expr: Expression) = aliasMap.getOrElseUpdate(expr, Alias(expr, expr.toString)())
+    def toAlias(expr: AggregateExpression) = aliasMap.getOrElseUpdate((expr.aggregateFunction.prettyName,expr.references.head), Alias(expr, expr.toString)())
```

Comment:
Suppose we execute this query "select count(*) from customer", the AttributeSet of this query will be empty and throw a java.util.NoSuchElementException.